js/src/common/utils/RequestError.ts

```diff
@@ -1,5 +1,14 @@
 export default class RequestError {
-  constructor(status, responseText, options, xhr) {
+  status: string;
+  options: object;
+  xhr: XMLHttpRequest;
+
+  responseText: string | null;
```

Comment:
How is this different from `responseText?: string;`?

Reply:
`?:` means this type or undefined. Undefined isn't treated the same as null.

Here we are saying responseText *will* be defined, just may be null.

Reply:
I think this is true only if the strict flag is enabled though, right? It's quite likely that I'm missing something, but I don't see it being set anywhere 🤔

Reply:
🤷 There _is_ a difference between undefined & null, so I think we should make it clear that it'll be defined but possibly null.

Reply:
Yes, but what I meant is that without strict mode you can still assign `null` to any variable or field even without explicitly writing that. To disallow this behaviour you need strict mode, which is usually recommended as a default configuration anyway.

Reply:
Ah I see what you mean. Not sure if we want strict mode because, due to the nature of core, most stuff isn't declared until the initializers run, and even then stuff like app.forum isn't set. It'll be set in most instances, but we can't really declare those.

Reply:
I'd love to try out strict mode, to be honest. But we can do that when we get to TypeScript-y-fying the application class.